File: dener8/programmers_2xn타일링.py
'''
점화식 세우기
1 2 3 5 ..
'''

def solution(n):
    li = [1, 2]
    if n == 1:
        return 1
    if n == 2:
        return 2
    for i in range(2, n):
        li.append((li[i - 1] + li[i - 2]) % 1_000_000_007)

    return li[-1]

File: myunghyunnero/programmers_2Xn타일링.cpp
#include <string>
#include <vector>

using namespace std;
int arr[60001];
int solution(int n) {
    int answer = 0;
    arr[1]=1;
    arr[2]=2;
    for(int i=3;i<=60000;i++){
        arr[i]=(arr[i-1]+arr[i-2])%1000000007;
    }
    return arr[n];
}